tests/functional/disaster-recovery/regional-dr/test_failover_and_relocate_discovered_apps.py

```diff
@@ -103,177 +117,179 @@ def test_failover_and_relocate_discovered_apps(
-        )
+            if primary_cluster_down:
+                config.switch_to_cluster_by_name(primary_cluster_name_before_failover)
+                logger.info(
+                    f"Stopping nodes of primary cluster: {primary_cluster_name_before_failover}"
+                )
+                nodes_multicluster[
+                    primary_cluster_name_before_failover_index
+                ].stop_nodes(primary_cluster_name_before_failover_nodes)
 
-        if primary_cluster_down:
-            logger.info(
-                f"Waiting for {wait_time} minutes before starting nodes "
-                f"of primary cluster: {primary_cluster_name_before_failover}"
-            )
-            sleep(wait_time * 60)
-            nodes_multicluster[primary_cluster_name_before_failover_index].start_nodes(
-                primary_cluster_name_before_failover_nodes
-            )
-            wait_for_nodes_status(
-                [node.name for node in primary_cluster_name_before_failover_nodes]
-            )
-            logger.info("Wait for 180 seconds for pods to stabilize")
-            sleep(180)
-            logger.info(
-                "Wait for all the pods in openshift-storage to be in running state"
+            dr_helpers.failover(
+                failover_cluster=secondary_cluster_name,
+                namespace=rdr_workload.workload_namespace,
+                discovered_apps=True,
+                workload_placement_name=rdr_workload.discovered_apps_placement_name,
+                old_primary=primary_cluster_name_before_failover,
             )
-            assert wait_for_pods_to_be_running(
-                timeout=720
-            ), "Not all the pods reached running state"
-            logger.info("Checking for Ceph Health OK")
-            ceph_health_check()
 
-        logger.info("Doing Cleanup Operations")
-        dr_helpers.do_discovered_apps_cleanup(
-            drpc_name=rdr_workload.discovered_apps_placement_name,
-            old_primary=primary_cluster_name_before_failover,
-            workload_namespace=rdr_workload.workload_namespace,
-            workload_dir=rdr_workload.workload_dir,
-            vrg_name=rdr_workload.discovered_apps_placement_name,
-        )
+            if primary_cluster_down:
+                logger.info(
+                    f"Waiting for {wait_time} minutes before starting nodes "
+                    f"of primary cluster: {primary_cluster_name_before_failover}"
+                )
+                sleep(wait_time * 60)
+                nodes_multicluster[
+                    primary_cluster_name_before_failover_index
+                ].start_nodes(primary_cluster_name_before_failover_nodes)
+                wait_for_nodes_status(
+                    [node.name for node in primary_cluster_name_before_failover_nodes]
+                )
+                logger.info("Wait for 180 seconds for pods to stabilize")
+                sleep(180)
```

Comment:
Please replace with a wait for resource function or equivalent

Reply:
I removed the sleep, as we already had wait_for_pods_running func below.